src/sentry/api/serializers/models/group.py

```diff
@@ -481,7 +481,7 @@ def _get_status(self, attrs, obj):
             status_label = "pending_merge"
         elif status == GroupStatus.REPROCESSING:
             status_label = "reprocessing"
-            status_details["pendingEvents"] = get_num_pending_events(attrs["id"])
+            status_details["pendingEvents"], status_details["info"] = get_progress(attrs["id"])
```

Comment:
We shouldn't be making io calls in this method. `_get_status` is called in `serialize`, which is called once per group. Ideally, you'd perform this logic in batch in `get_attrs` and just fetch the value from `attrs` here. We probably shouldn't have had `get_num_pending_events` here to begin with, since that would also have been making an io call per group, but now we will be making 2 per reprocessed group, which makes this endpoint slower.

So what you could do is build `get_progress_for_groups` that accepts a list of group_ids, then do something like

```
    keys = []
    for group_id in group_ids:
        keys.append(_get_sync_counter_key(group_id))
        keys.append(_get_info_reprocessed_key(group_id))
    results = client.mget(keys)
    # unpack results
```

Then just call that in `get_attrs` and store the results for each group in `attrs`.

Reply:
Damn, just read the `redis-py-cluster` code and it just performs `mgets` in serial :( https://github.com/Grokzen/redis-py-cluster/blob/master/rediscluster/client.py#L985. 

Unblocking for now, although it's not great that we're doing this many IO calls here, not sure whether there's something we can do to improve this later on.

Reply:
I'm fine with this for now as we expect there will just not be a lot of reprocessing issues at once, and because is:reprocessing is a transient state.

Didn't know about the mget impl, that's messed up. It does not even do grouped mget to the same node so we can't optimize at all